lib/src/protobuf/json.dart

```diff
@@ -5,6 +5,14 @@
 part of protobuf;
 
 Map<String, dynamic> _writeToJsonMap(_FieldSet fs) {
+  return _writeToJsonMapWithMapper(fs, (fi) => '${fi.tagNumber}');
```

Comment:
I think we might have to inline the _writeToJsonMapWithMapper in each case to make sure we do not degrade performance. @rakudrama do you agree?

Reply:
The main issue is how to get the recursive call at line 232 to encode/decode nested messages with the same policy for property names. I believe this CL is broken for nested structures.

Passing through a flag might be the best option.

Secondarily, I don't really like parameterizing with functions. Usually this leads to an allocation at the function creation site, and the indirect call has similar performance to an if-check.